.github/workflows/standalone.yml

```diff
@@ -0,0 +1,429 @@
+      name: Build (MSBuild)
+      working-directory: ./leveldb/build
+      run: msbuild ./leveldb.sln /p:Configuration=Release
+
+    - if: ${{ steps.cache-leveldb.outputs.cache-hit != 'true' }}
+      name: Cache Distribution
+      uses: actions/cache/save@v3
+      with:
+        path: ./leveldb/build/Release/*
+        key: leveldb-windows-${{ matrix.arch }}
+        enableCrossOsArchive: true
+
+  build-neo-cli:
+    needs: [build-leveldb]
+    name: Build TarBalls (${{ matrix.runtime }})
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        runtime: [linux-x64, linux-arm64, linux-musl-x64, linux-musl-arm64, win-x64, win-arm64]
+
+    steps:
+    - name: Checkout Repository Code
+      uses: actions/checkout@v4
+      with:
+        path: neo
+        fetch-depth: 0
+
+    - name: Checkout Repository Code (neo-modules)
+      uses: actions/checkout@v4
+      with:
+        repository: neo-project/neo-modules
+        path: neo-modules
+        fetch-depth: 0
+
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v4
+      with:
+        dotnet-version: ${{ env.DOTNET_VERSION }}
+
+    - name: .NET Publish (neo-cli)
+      working-directory: ./neo
+      run: |
+        dotnet publish ./src/Neo.CLI \
+        --framework net7.0 \
+        --configuration ${{ env.CONFIGURATION }} \
+        --runtime ${{ matrix.runtime }} \
+        --self-contained true \
+        --output ${{ env.OUTPUT_PATH }}/${{ matrix.runtime }} \
+        --verbosity normal \
+        -p:RuntimeIdentifier=${{ matrix.runtime }} \
+        -p:SelfContained=true \
+        -p:IncludeNativeLibrariesForSelfExtract=false \
+        -p:PublishTrimmed=false \
+        -p:PublishSingleFile=true \
+        -p:PublishReadyToRun=true \
+        -p:EnableCompressionInSingleFile=true \
+        -p:DebugType=embedded \
+        -p:ServerGarbageCollection=true
+
+    - name: .NET Build (LevelDBStore)
+      working-directory: ./neo-modules
+      run: |
+        dotnet build ./src/LevelDBStore/LevelDBStore.csproj \
+        --framework net7.0 \
+        --configuration ${{ env.CONFIGURATION }} \
+        --output ${{ env.OUTPUT_PATH }}/${{ matrix.runtime }}/Plugins/LevelDBStore \
+        --verbosity normal
+
+    - name: Remove Xml Comment Files
+      working-directory: ${{ env.OUTPUT_PATH }}/${{ matrix.runtime }}
+      run: rm -v *.xml
+
+    - if: ${{ startsWith(matrix.runtime, 'win-x64') }}
+      name: Get Distribution Caches (win-x64)
+      uses: actions/cache@v3
+      with:
+        path: ./leveldb/build/Release/*
+        key: leveldb-windows-x64
+        enableCrossOsArchive: true
+        fail-on-cache-miss: true
+
+    - if: ${{ startsWith(matrix.runtime, 'win-arm64') }}
+      name: Get Distribution Caches (win-arm64)
+      uses: actions/cache@v3
+      with:
+        path: ./leveldb/build/Release/*
+        key: leveldb-windows-arm64
+        enableCrossOsArchive: true
+        fail-on-cache-miss: true
```

Comment:
Move `key` to matrix parameter, exactly as in https://github.com/neo-project/neo-devpack-dotnet/pull/855#discussion_r1535555724, and then two steps may be squashed. Cleaner code, no copy-pasting. The same optimisation can be applied to the instructions below.